### ISSUE TITLE
Completely broken

### ISSUE THREAD

**Issue by MPX4132** [FluxionNetwork/fluxion#881]
Code has been deleted where it shouldn't have been.
The DNS python script isn't even in the freaking repo anymore; what the hell?
I wouldn't be surprised if the script no longer works.

**Comment by xpz3** [comment]
https://github.com/FluxionNetwork/fluxion/commit/4bf312be4ac2f6b199e63bf6eb3464d956f533f7

Just revert this commit

**Comment by xpz3** [comment]
If you revert that commit the following should ve reverted too
https://github.com/FluxionNetwork/fluxion/commit/db7a5e905d99ed5485aa3f91b3f43cf2984e30fc

https://github.com/FluxionNetwork/fluxion/commit/bf8a6537cb6b5dc0bcea33403e94f1bb4746966e

https://github.com/FluxionNetwork/fluxion/commit/4cddfb6c25001d11f2ff85fe260504fd9efbeee5

**Comment by MPX4132** [comment]
I'll mess with it later, I'm going to bed. It's 4AM US central time, and I'm pretty sleepy.

**Comment by strasharo** [comment]
It's using dnsspoof now.

**Comment by xpz3** [comment]
> It's using dnsspoof now.

It was done to easily use a hosts file for custom redirection

**Comment by MPX4132** [comment]
If I recall correctly, I didn't use that because iOS doesn't pick it up immediately. Literally all Google sites, threw out all Apple support. That's probably why someone complained about the code what's his name added.

**Comment by xpz3** [comment]
> If I recall correctly, I didn't use that because iOS doesn't pick it up immediately. Literally all Google sites, threw out all Apple support. That's probably why someone complained about the code what's his name added.

I don't quite get the meaning of last line. Do you mean someone complained about fluxion? Or did you mean a guy saying he couldn't get captive portal load on his xiaomi device? If yes, then it's not directly related to dnsspoof being used. And for me when i did tests with different iOS devices, they did open captive portal without any problem until now. 
But yeah there were some peculiar cases that popped up here in github issues or in the discord server. But that wasn't directly related to dnsspoof.
For example the the first guy that came to discord was complaining that the captive portal gets loaded on all his devices after many page refreshes only. The portal does get detected but failed to load the page, even though all relevant services like lighttpd dns etc were up and running. I was able to solve it by disabling the code to redirect all sites to captive.gateway.lan in lighttpd. It worked for him.

**Comment by kandarpgautam** [comment]
yea. currently it's completely broken. nothing is working.

**Comment by kandarpgautam** [comment]
@MPX4132 please keep the master branch stable and clean!! 
fluxion also fetches updates from master branch automatically. which might even cause issues to users which had no problem. i recommend to make development branch separate from master branch. only merge stable release in master branch.

**Comment by xpz3** [comment]
> @MPX4132 please keep the master branch stable and clean!!
> fluxion also fetches updates from master branch automatically. which might even cause issues to users which had no problem. i recommend to make development branch separate from master branch. only merge stable release in master branch.

Its good to always have a dev branch. But i don't think fluxion is completely broken now. Majority of users has it working. In kali, the dependency pyrit was removed such should be manually installed to get fluxion up and running. I think you are good to go from there. If you have some specific issue, you can ask by opening an issue here or join fluxion discord and requesting for help there. But by saying nothing is working, you are just making the devs confusing. I recommend you post your specific issue in discord. Maybe it's not even related to fluxion. It could be your chipset, your Linux env, or even a bug in fluxion itself but when reporting please be specific..

**Comment by usama7628674** [comment]
Kimocoder's fork of pyrit is ported to python 3.I should open a ticket on kali bugs site and let them know about this fork so they may add it in kali-rolling.
https://github.com/kimocoder/Pyrit

**Comment by MPX4132** [comment]
Fixed some bugs to get the script working again. The authenticator was the point of failure. Cowpatty wasn't working, so I added options for the tool to be used during authentication. There's still more bugs, but they're with other tools. I saw mdk4 crash multiple times while I was testing, I saw the DNS spoofer crash multiple times, and also saw the access point crash once only, surprisingly. That used to be the flaky part.
